src/main/java/org/jabref/gui/contentselector/ContentSelectorDialogView.java

```diff
@@ -0,0 +1,125 @@
+package org.jabref.gui.contentselector;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import javafx.beans.property.ListProperty;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.ListView;
+import javafx.scene.control.SelectionModel;
+
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.ControlHelper;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.metadata.MetaData;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import org.fxmisc.easybind.EasyBind;
+
+public class ContentSelectorDialogView extends BaseDialog<Void> {
+
+    @FXML
+    private Button addFieldNameButton;
+    @FXML
+    private Button removeFieldNameButton;
+    @FXML
+    private Button addKeywordButton;
+    @FXML
+    private Button removeKeywordButton;
+    @FXML
+    private ListView<String> fieldNamesListView;
+    @FXML
+    private ListView<String> keywordsListView;
+    @FXML
+    private ButtonType saveButton;
+
+    private ContentSelectorDialogViewModel viewModel;
+
+    private final BasePanel basePanel;
+    private final DialogService dialogService;
+    private final MetaData metaData;
+
+    public ContentSelectorDialogView(JabRefFrame jabRefFrame) {
+        this.setTitle(Localization.lang("Manage content selectors"));
+        this.getDialogPane().setPrefSize(375, 475);
+
+        this.basePanel = jabRefFrame.getCurrentBasePanel();
+        this.dialogService = jabRefFrame.getDialogService();
```

Comment:
I would change the constructor to only accept basePanel as Argument, because you can do the following:
    @Inject private DialogService dialogService;
This automatically injects (passes an instance of it) to the class

Reply:
Just this one change and it looks good to me ;)

Reply:
Good point :) fixed - c32fe6a